### ISSUE TITLE
Connector fails during initialization - Invalid property path for getNestedPropertyDefinition

### ISSUE THREAD

**Issue by SippieCup** [strongloop/loopback-connector-postgresql#364]
It seems the newest pull request in loopback-connector (https://github.com/strongloop/loopback-connector/pull/136) breaks the postgreql connector.

# Description/Steps to reproduce

npm install, then run any psql connected application. 

# Expected result
```
NODE_ENV=production /opt/elasticbeanstalk/node-install/node-v10.15.0-linux-x64/bin/node .
/var/app/current/node_modules/loopback-connector/lib/connector.js:159
    throw new Error(g.f('Invalid property path'));
    ^

Error: Invalid property path
    at Function.Connector.getNestedPropertyDefinition (/var/app/current/node_modules/loopback-connector/lib/connector.js:159:11)
    at PostgreSQL.Connector.getPropertyDefinition (/var/app/current/node_modules/loopback-connector/lib/connector.js:139:20)
    at PostgreSQL.SQLConnector.column (/var/app/current/node_modules/loopback-connector/lib/sql.js:376:19)
    at /var/app/current/node_modules/loopback-connector/lib/sql.js:157:27
    at Array.forEach (<anonymous>)
    at PostgreSQL.SQLConnector.searchForPropertyInActual (/var/app/current/node_modules/loopback-connector/lib/sql.js:156:16)
    at /var/app/current/node_modules/loopback-connector-postgresql/lib/migration.js:135:24
    at Array.forEach (<anonymous>)
    at PostgreSQL.getColumnsToAdd (/var/app/current/node_modules/loopback-connector-postgresql/lib/migration.js:133:15)
    at PostgreSQL.getAddModifyColumns (/var/app/current/node_modules/loopback-connector-postgresql/lib/migration.js:117:27)
    at applyPending (/var/app/current/node_modules/loopback-connector-postgresql/lib/migration.js:86:45)
    at /var/app/current/node_modules/loopback-connector-postgresql/lib/migration.js:104:9
    at /var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:607:21
    at /var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:246:17
    at iterate (/var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:146:13)
    at async.eachSeries (/var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:162:9)
    at _asyncMap (/var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:245:13)
    at Object.mapSeries (/var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:228:23)
    at Object.async.series (/var/app/current/node_modules/loopback-connector-postgresql/node_modules/async/lib/async.js:605:19)
    at PostgreSQL.alterTable (/var/app/current/node_modules/loopback-connector-postgresql/lib/migration.js:102:11)
    at /var/app/current/node_modules/loopback-connector/lib/sql.js:86:14
    at /var/app/current/node_modules/loopback-connector/lib/sql.js:249:16
```

Workaround: Explictly installing loopback-connector@4.5.1 fixes the issue, but is obviously not ideal

**Comment by vidmar** [comment]
Same here! Broke all my builds.

**Comment by dhmlau** [comment]
@b-admike, it might be related to your latest change. Could you please take a look? Thanks.

**Comment by b-admike** [comment]
Thank you for bringing this to our attention @SippieCup @vidmar. I agree that we've changed the API in terms of throwing an error for first level properties that are not defined. I've made a fix to make the changes backward compatible. Once it is landed, we'll release a new version, but until then workaround sounds good. Sorry for the inconvenience, I should not have merged it before the downstream tests ran (it was stuck for some reason over the weekend).

**Comment by b-admike** [comment]
Cross-posting from https://github.com/strongloop/loopback-connector/pull/138#issuecomment-465643402

> > I'm not familiar with the postgres connector, and why is trying to access non-defined property definition.
> 
> For trying to access undefined property definitions, there is a bug in the connector which I think I have a fix at https://github.com/strongloop/loopback-connector-postgresql/pull/365. But, we also returned the property definition as is before (even if it was undefined), so in some cases it might be working as expected. 
> 
> > The solution proposed looks good, I wonder if we should return undefined even for nested properties, for consistency.
> 
> Yes that would be better.
> 
> > PS: Honestly I feel that throwing an exception in this case is correct, but I understand that it's not backward compatible.
> 
> I agree with this. I have pulled the released version `4.6.0` back so we can assess how to proceed with #136 for release. I think we should release it as a semver major since it has the breaking changes (I feel we can improve on the error message depending on the condition, but that can be done in a follow-up PR). If we decide to go ahead with this patch, then we can release a new version which is a semver minor and still backward-compatible. @strongloop/ch-lb-db-connectors @strongloop/sq-lb-apex Thoughts?
> 
> _Originally posted by @b-admike in https://github.com/strongloop/loopback-connector/pull/138#issuecomment-465643402_
